src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -88,13 +90,19 @@ public void save(AddressBook addressBook) throws StorageOperationException {
         /* Note: Note the 'try with resource' statement below.
          * More info: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html
          */
+        
+        if (filePathCheck() && fileCheck){
```

Comment:
this is a good place to check, I agree.
however, you haven't used exception mechanism as required by the assignment :-P 
think how you can use exception as a mechanism to show the message to the user and continue to do the operation undisturbed.

no need to re-submit a PR for this. I just want you to try and understand the mechanism :-)